src/client/containers/CartPage/CartPage.Container.js

```diff
@@ -66,54 +93,89 @@ const CartPageContainer = ({ isAuthenticated }) => {
     const productId = cartItem[index].productId;
     changeProductQuantity(productId, 0);
   };
+
+  const checkFavortie = (productId) => {
+    if (favorite.length > 0) {
+      return (
+        favorite.filter((item) => item.id === parseInt(productId)).length > 0
+      );
+    }
+    return false;
+  };
+
+  const showCartItems = () => {
+    if (cartItem.length === 0) {
+      return <h1> Your Cart is Empty </h1>;
+    }
+
+    return cartItem.map((item, index) => {
+      return (
+        <ShoppingItem
+          productId={item.productId}
```

Comment:
It is good to add **key** to this map, for preventing warning in console.

Reply:
accepted. done